### DISCUSSION TITLE
Help to contributing

### DISCUSSION THREAD

**Discussion by Horsty80** [codeceptjs/CodeceptJS#3736] [Q&A]
Hello :) 

I want to help on this repo, but when i fork the repo and try to install dependencies isn't working...

I see nothing in contribute guide about this.

What version of nodejs is needed ?
What package manager is needed (npm / pnpm / yarn) ? there are no lock file it's normal ?

Thanks :)

**Comment by kobenguyent** [comment]
I don't think it's quite strict to force you use a specific node or package management, based on readme, CodeceptJS requires Node.js version 12+ or later, and package management should be your prefer.

**Comment by Horsty80** [comment]
I'm agree with that :) 

Hum i'm using pnpm and i'm facing issue with electron.... 

I'm using a macbook pro M1 ultra, maybe it's that
node v18.16.1
pnpm v8.5.0

<img width="1250" alt="Capture d’écran 2023-07-05 à 13 29 23" src="https://github.com/codeceptjs/CodeceptJS/assets/15615569/8a5570a9-2fb9-40b5-845d-dff9cbd398c2">

https://github.com/electron/electron/releases/download/v2.0.18/electron-v2.0.18-darwin-arm64.zip is in 404

**Comment by Horsty80** [comment]
I've found, it's due to ARM architecture of M1...

More information here: https://www.electronjs.org/fr/blog/apple-silicon

I'll check if i get a solution about this

**Reply by kobenguyent** [reply]
oh weird, I'm also using M1 Macbook and works fine with npm.

**Reply by Horsty80** [reply]
It's my personnal laptop, so no proxy or strange configuration ^^', i don't understand why i'm facing that...

**Reply by kobenguyent** [reply]
may you try npm or yarn to see if that's also an issue

**Reply by Horsty80** [reply]
Yes i'll try later i'm going to work ^^ thanks for your help @kobenguyent 

I'm using this lib very often, so i hope i can help you by contributing :)

**Reply by kobenguyent** [reply]
```

.../node_modules/electron postinstall$ node install.js
Downloading electron-v2.0.18-darwin-arm64.zip
│ Error: GET https://github.com/electron/electron/releases/download/v2.0.18/electron-v2.0.18-darwin-arm64.zip returned 404
│ /Users/thanh.nguyen/Desktop/CodeceptJS/node_modules/.pnpm/electron@2.0.18/node_modules/electron/install.js:54
│   throw err
│   ^
│ Error: Failed to find Electron v2.0.18 for darwin-arm64 at https://github.com/electron/electron/releases/download/v2.0.18/electron-v2.0.18-darwin-arm64.zip
│     at Request.<anonymous> (/Users/thanh.nguyen/Desktop/CodeceptJS/node_modules/.pnpm/nugget@2.2.0/node_modules/nugget/index.js:197:61)
│     at Request.emit (node:events:513:28)
│     at Request.onRequestResponse (/Users/thanh.nguyen/Desktop/CodeceptJS/node_modules/.pnpm/request@2.88.2/node_modules/request/request.js:1059:10)
│     at ClientRequest.emit (node:events:513:28)
│     at HTTPParser.parserOnIncomingClient [as onIncoming] (node:_http_client:701:27)
│     at HTTPParser.parserOnHeadersComplete (node:_http_common:119:17)
│     at TLSSocket.socketOnData (node:_http_client:542:22)
│     at TLSSocket.emit (node:events:513:28)
│     at addChunk (node:internal/streams/readable:324:12)
│     at readableAddChunk (node:internal/streams/readable:297:9)
│ Node.js v18.16.0
└─ Failed in 1.4s at /Users/thanh.nguyen/Desktop/CodeceptJS/node_modules/.pnpm/electron@2.0.18/node_modules/electron
node_modules/.pnpm/highlight.js@9.18.5/node_modules/highlight.js: Running postinstall script, done in 105ms
node_modules/.pnpm/playwright@1.35.1/node_modules/playwright: Running install script...
 ELIFECYCLE  Command failed with exit code 1.

```
looks like the issue with pnpm @Horsty80

**Reply by Horsty80** [reply]
It's look exactly the same indeed, electron-v2.0.18-darwin-arm64.zip

**Reply by kobenguyent** [reply]
I think I found the issue, it's actually the nightmarejs which is using electron v2.0.18
https://github.com/segmentio/nightmare/blob/master/package.json#L24
